### ISSUE TITLE
spdlog colors with MinGW/Cygwin

### ISSUE THREAD

**Issue by LunarWatcher** [gabime/spdlog#950]
TL;DR: Color logging doesn't work. It compiles fine, but doesn't work.

Longer explanation:

Started out with this (imports omitted):

```cpp
int main() {
    auto console = spdlog::stdout_color_mt("console");
    console->warn("Testing");
}
```

It didn't print anything. I also had a print statement (`std::cout`) after, which printed. After a bunch of failed attempts, changing import method (from just `includes` to the entire project and using CMake's `add_subdirectory` to compile it), I managed to find issue 492.

Following [492](https://github.com/gabime/spdlog/issues/492), I ended up with this:

```cpp
int main() {
    auto sink = std::make_shared<spdlog::sinks::ansicolor_stdout_sink_mt>();
    auto console = std::make_shared<spdlog::logger>("console", sink);
    console->warn("Testing");
}
```

I ran it in the three bash terminals I have available, just to see if there's a problem with any of those. The result is the same in all three: no color. 

Using the first code snippet and running from Powershell or cmd works fine; there's color. Using the first in Cygwin or MinGW prints nothing. Using the second in Cygwin or MinGW prints, but no color. Using cmd or Powershell prints `[2018-12-28 18:22:17.625] [console] [[33m[1mwarning[m] Testing`; the data for the colors are clearly there, but it still doesn't work. 

If it matters, technical details:

* OS: Win10
* Compiler: GCC
* Build system: CMake

---

Also stumbled over [this](https://stackoverflow.com/questions/2048509/how-to-echo-with-different-colors-in-the-windows-command-line#comment70633610_38617204) on Stack Overflow. I ran the script in Cygwin, MinGW, Powershell, and CMD, and it showed color in all of them. I haven't tested on linux, but I assume the standard is the same. However, I can't verify if it works there or not. However, I suspect there might be a bug in the output creation.

**Comment by gabime** [comment]
Seems like a duplicate of issue #536

**Comment by LunarWatcher** [comment]
Except that uses wincolor. I explicitly use the ANSI sinks on the last attempt. It makes sense the wincolor sinks don't work in bash, because it's a different terminal. What doesn't make sense is ANSI not working. Also, digging into [the source](https://github.com/gabime/spdlog/blob/v1.x/include/spdlog/sinks/wincolor_sink.h#L120), wincolor uses a native Windows call rather than using specific output codes. See also [this](https://superuser.com/a/184355/623128) on Superuser; The Windows API calls can't be used for color on Cygwin/MinGW/related. However, it not showing up with the ANSI sinks explicitly used seems like a bug

**Comment by gabime** [comment]
>  it not showing up with the ANSI sinks explicitly used 

Not showing color? or not showing at all?

**Comment by LunarWatcher** [comment]
With the explicit ANSI sink, it only fails to show color. The text itself shows. ANSI should work with CMD too (Win10 CMD supports it). With the wincolor sink, nothing shows at all outside CMD (which is kinda expected since it uses an unsupported API with Cygwin).

**Comment by gabime** [comment]
Probably because it detects it as colorless or not terminal (
https://github.com/gabime/spdlog/blob/v1.x/include/spdlog/sinks/ansicolor_sink.h#L41)

Could you quick debug it and see if this is the case?

**Comment by LunarWatcher** [comment]
`in_terminal` returns false, `is_color_terminal` returns true.

**Comment by gabime** [comment]
@LunarWatcher There is a new flag to force color even if no color terminal is detected. 
Please try the following and feedback if it helps:
 ```c++
color_sink->set_color_mode(spdlog::color_mode::always::always);
```
